tgui/packages/tgui/interfaces/PersonalCooking.tsx

```diff
@@ -0,0 +1,586 @@
+                        activeCategory === category && searchText.length === 0
+                      }
+                      onClick={(e) => {
+                        setCategory(category);
+                        document.getElementById('content').scrollTop = 0;
+                        if (searchText.length > 0) {
+                          setSearchText('');
+                        }
+                      }}>
+                      <Stack>
+                        <Stack.Item width="14px" textAlign="center">
+                          <Icon name={CATEGORY_ICONS[category]} />
+                        </Stack.Item>
+                        <Stack.Item grow>{category}</Stack.Item>
+                        {category === 'Can Make' && (
+                          <Stack.Item>
+                            {Object.keys(data.craftability).length}
+                          </Stack.Item>
+                        )}
+                      </Stack>
+                    </Tabs.Tab>
+                  ))}
+              </Tabs>
+              <hr style={{ 'border-color': '#111' }} />
+              <Button.Checkbox
+                fluid
+                content="Can make only"
+                checked={data.display_craftable_only}
+                onClick={() => {
+                  data.display_craftable_only = !data.display_craftable_only;
+                  act('toggle_recipes');
+                }}
+              />
+              <Button.Checkbox
+                fluid
+                content="Compact list"
+                checked={data.display_compact}
+                onClick={() => act('toggle_compact')}
+              />
+            </Section>
+          </Stack.Item>
+          <Stack.Item grow>
+            {recipes.length > 0 ? (
+              recipes
+                .slice(0, displayLimit)
+                .map((item) =>
+                  data.display_compact ? (
+                    <RecipeContentCompact
+                      key={item.ref}
+                      item={item}
+                      craftable={Boolean(craftabilityByRef[item.ref])}
+                      busy={data.busy}
+                    />
+                  ) : (
+                    <RecipeContent
+                      key={item.ref}
+                      item={item}
+                      diet={data.diet}
+                      craftable={Boolean(craftabilityByRef[item.ref])}
+                      busy={data.busy}
+                    />
+                  )
+                )
+            ) : (
+              <NoticeBox m={1} p={1}>
+                No recipes found.
+              </NoticeBox>
+            )}
+            {recipes.length > displayLimit && (
+              <Section textAlign="right">
+                And {recipes.length - displayLimit} more...
+              </Section>
+            )}
+          </Stack.Item>
+        </Stack>
+      </Window.Content>
+    </Window>
+  );
+};
+
+const TypeContent = (props) => {
+  return (
```

Comment:
not bad, but not great. we should be destructuring props either immediately (in the arguments) or on the next line if you would like to keep the function signature intact